dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/transport/AbstractClient.java

```diff
@@ -179,6 +179,12 @@ protected void connect() throws RemotingException {
 
         try {
 
+            if (isClosed()) {
```

Comment:
in case of reconnect would it create any issue? as reconnect call the connect method?

Reply:
The client can't be reused, it can only be closed once. So, if you decide to close a client, then when you close it, even if this is reconnecting the client, you can't make reconnect successful.

client其实是不能复用的，只能关闭一次，所以，如果决定关闭一个client，那么将其close时，哪怕这是正在对这个client进行reconnect，也不能让reconnect成功。